File: src/main/java/com/mytrip/demo/domain/Trip.java
package com.mytrip.demo.domain;

import lombok.Builder;
import lombok.Value;

import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

@Builder
@Value
public class Trip {
    UUID uuid;
    String title;
    String creatorEmail;
    LocalDate from;
    LocalDate to;
    List<String> participantsEmails;
    List<Trip> trips;
}

File: frontend/src/app/trip/model/Trip.ts
import {Event} from "./Event";

export interface Trip {
  uuid: string;
  title: string;
  creatorEmail: string;
  from: string;
  to: string;
  participantsEmails: string[];
  events: Event[];
}

export interface TripsResponse {
  trips: Trip[];
}

export interface CreateTripRequestModel {
  title: string | null;
  from: string | null;
  to: string | null;
}

export interface UpdateTripRequestModel {
  uuid: string | null;
  title: string | null;
  from: string | null;
  to: string | null;
}

export interface AddParticipantModel {
  email: string | null;
  uuid: string | null;
}
